### ISSUE TITLE
ThingsBoard generates a large number of PostgreSQL idle processes

### ISSUE THREAD

**Issue by fabio1299** [thingsboard/thingsboard#2363]
Hi,
I have ThingsBoard running on a VMware VM under Ubuntu 18.04 server.
Data is stored on a PostgreSQL 11.6 running on a different VM also under Ubuntu 18.04 server.
Things are working fine, but I've noticed up to 50 process originating from the ThingsBoard server and accessing the PostgreSQL database. And all of them show as being idle.

Device data is submitted to ThingsBoard using the HTTP through a NODE-RED server.
 
Any idea on what's going on?

Many thanks,
Fabio

**Comment by bbedwell** [comment]
I was just noticing the same thing and came here to investigate. It’s from the node service, and it seems very excessive to have 50 connections open, mostly staying idle. It makes it difficult or even impossible to scale or plan for redundancy when each process uses that many connections and then exhausts the available connections in PostgreSQL.

Is this only in the latest version(s)?

I’ll look forward to anyone else’s thoughts.

**Comment by hamburml** [comment]
I also see a lot of idle sessions. If I use a flot widget and want to display data of the last 30 days it takes really long to load. I checked the server load via top and the server is mainly idling.
I don't know if Thingsboard or Postgresql is the culprit.

May I request a comment @ashvayka?

**Comment by wang701** [comment]
+1

**Comment by bazenkov** [comment]
I have got the same behavior. 
I examined the connections by the query:
`SELECT state,wait_event_type,substring(query,1,100) FROM pg_stat_activity WHERE datname='thingsboard';`

It shows that the wait_event_type is mostly 'Client' which means the database server waits for an action by the application.
This behavior was observed in TB 2.5.4 and now I've just updated to 3.2 and the situation is the same. Seems like some connection leak occurs in TB.
